GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -1707,13 +1758,107 @@ private void writeStatsAndGraphsSheet(
         }
     }
 
-    /**
-     * creates a cell in the row passed in and sets the style and value (if non-null)
-     */
+    private int writeCollectionStats(Map<QuestionGroupDto,
+            List<QuestionDto>> questionMap,
+            Sheet sheet,
+            int curRow) {
+        final int tagCol = 0;
+        final int valCol = 3;
+        //Global statistics
+        //Calculate them first (due to the window, we cannot sum this while looping and 
+        // then go back up and draw it last)
+        int totalQuestions = 0;
+        for (List<QuestionDto> group : questionMap.values()) {
+            totalQuestions += group.size();
+        }
+        //Now draw them
+        curRow++;
+        Row statRow = getRow(curRow++, sheet);
+        createCell(statRow, tagCol, "Questions");
+        createCell(statRow, valCol, totalQuestions);
+   
+        statRow = getRow(curRow++, sheet);
+        createCell(statRow, tagCol, "Form submissions");
+        createCell(statRow, valCol, totalInstances);
+        
+        if (totalInstances == 0) return curRow + 2; //add a little space
+   
+        //The following two cells could also be made into date cells
+        statRow = getRow(curRow++, sheet);
+        createCell(statRow, tagCol, "First submission");
+        createCell(statRow, valCol, ExportImportUtils.formatDateTime(firstSubmission));
+   
+        statRow = getRow(curRow++, sheet);
+        createCell(statRow, tagCol, "Last submission");
+        createCell(statRow, valCol, ExportImportUtils.formatDateTime(lastSubmission));
+        
+        statRow = getRow(curRow++, sheet);
+        createCell(statRow, tagCol, "Shortest duration");
+        createCell(statRow, valCol, getDurationText(minDuration));
+   
+        statRow = getRow(curRow++, sheet);
+        createCell(statRow, tagCol, "Longest duration");
+        createCell(statRow, valCol, getDurationText(maxDuration));
+   
+        statRow = getRow(curRow++, sheet);
+        createCell(statRow, tagCol, "Average duration");
+        try {
+            createCell(statRow, valCol, getDurationText((long) totalDuration / totalInstances));
+        } catch (Exception e) {
+            // swallow exception, leave cell empty
+        }
+   
+        for (String user : instancesByUser.keySet()) {
+            statRow = getRow(curRow++, sheet);
+            if (user != null) {
+                createCell(statRow, tagCol, "User " + user);
+            } else {
+                createCell(statRow, tagCol, "No user");
+            }
+            createCell(statRow, valCol, instancesByUser.get(user));   
+        }
+        
+        for (String dev : instancesByDevice.keySet()) {
+            statRow = getRow(curRow++, sheet);
+            createCell(statRow, tagCol, "Device " + dev);
+            createCell(statRow, valCol, instancesByDevice.get(dev));            
+        }
+        
+        return curRow + 2; //add a little space
+    }
+
+    
+    // Create a string cell
```

Comment:
why not call this createStringCell and then you can remove the comment

Reply:
Because it does the same thing, just for different argument types. We have lots of these in Flow.